esp-hal/src/ledc/timer.rs

```diff
@@ -149,7 +149,7 @@ impl TimerSpeed for HighSpeed {
 }
 
 /// Interface for Timers
-pub trait TimerIFace<S: TimerSpeed> {
+pub trait TimerIFace<S: TimerSpeed>: Sync {
```

Comment:
I don't think explicity requiring `Sync` is a good idea. The implementation is not thread-safe at all.

Reply:
In the latest commit, I have modified the `Timer` struct to consist of a single timer number, so that it was `Sync`. I believe adding `Sync` functionality is beneficial, as the current implementation involves a channel holding a reference to the timer, which prevents updating the timer while the reference is in use. Being able to update the timer on the fly—such as adjusting the frequency of the LEDC while the channel holds a reference—is a common and practical use case. Could you please review my pull request again at your convenience?

@bugadani 

```rust
/// Timer struct
#[derive(Clone, Copy)]
pub struct Timer<S: TimerSpeed> {
    phantom: PhantomData<S>,
    number: Number,
}
```